### PR TITLE
Use SphericalManifold for push_forward and pull_back

### PR DESCRIPTION
**Pull request by anne-glerum**
This way the push and pull functions don't have to be implemented again, and there is no problem with crossing the 180 hemisphere.

### CODE REVIEW

File: source/geometry_model/chunk.cc [hunk 1]
```diff
@@ -175,7 +117,6 @@ namespace aspect
       return std::map<std::string,types::boundary_id>();
     }
 
-
```
Comment (bangerth) [inline source/geometry_model/chunk.cc]: Oh, are you willing to re-add these empty lines? We typically keep 3 empty lines between functions.

**Comment by bangerth** [comment]
Nice simplification!

**Comment by bangerth** [comment]
There are 4 tests that fail because the velocities (and consequently other quantities) are now significantly different. Do you know why that is?

**Comment by anne-glerum** [comment]
I think the difference arrives from the original chunk model having pull and push  functions taking latitudes, while the SphericalManifold pull and push funtions take colatitude. Will correct for this.

**Comment by gassmoeller** [comment]
Closed by #1533

**Pull request closed without merging**